### PR TITLE
feat: dynamic OG images for docs #986

### LINKED ISSUE #986: feat(docs): dynamic og-images
### Is your feature request related to a problem? Please describe.

This week we rolled out dynamic og images for trpc: https://twitter.com/ixahmedxii/status/1603517506525732881

CleanShot 2022-12-16 at 20 58 38

Would be nice with something similar for our docs too

### Describe the solution you'd like to see

Dynamic OG Images for Documentation Pages

Things we can include:

- Title
- Description
- Language
- ???

Could also be a good time to change up the logo to the new one Theo announced on stream this week: https://t3.gg, https://github.com/t3dotgg/t3-astro/blob/main/public/images/t3/700.svg

### Describe alternate solutions

Drawback is it has some pitfalls in frameworks that's not Next.js: https://twitter.com/jullerino/status/1602346699359412225

Which might require us to do something similar to what we did in trpc and create a separate app that hosts it: https://github.com/trpc/trpc/tree/main/www/og-image

### Additional information

It was a bigger job than we first anticipated but here are some useful resources:

https://github.com/trpc/trpc/pull/3382
https://github.com/trpc/trpc/pull/3402
https://github.com/trpc/trpc/pull/3406

https://github.com/vercel/satori

### PR DESCRIPTION
**Pull request by iAverages**
Adds dynamic OG images for documentation. 

Closes #986

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

- Setup Astro for server builds
- Setup route for generating dynamic images using Satori

## Screenshots

English:
home
intro
image
image

Large title (Title font gets reduced to avoid overflow and word breaking)
image

Other LFR Languages:
spanish
french
pt
russian

RTL Languages:
image
image
image

💯

### CODE REVIEW

**Comment by iAverages** [comment]
OG generation is working. I need to change the design of the OG images themselves. I plan to make the title bold and play around with the sizing + logo color. If anyone has any ideas on what we can use please share!

_The red boxes are apart of debug mode in satori_
OG Example

**Comment by iAverages** [comment]
Enabling SSR with Astro means, by default, every page will be server-side rendered. Currently, when SSR is enabled prerendering pages (static page generation) is set to false for every page, with an opt-in variable `export const prerender = true`. Using SSR also means the automatic sitemap generation does not work. 

On Astro's roadmap, they have plans to change the default to allow for prerendering by default, with opt-in for SSR. https://github.com/withastro/roadmap/issues/539

If you guys do not want SSR then it would be best to wait for the Astro team to release the opt-in SSR per page.

**Comment by iAverages** [comment]
This PR is ready to merge, however, I am not sure if this should be merged yet due to the previous comment.

File: www/src/config.ts
Comment (juliusmarminge) [inline www/src/config.ts:25]:
```suggestion
```

File: www/src/utils/ogFont.ts
Comment (juliusmarminge) [inline www/src/utils/ogFont.ts:1]: looks like you snagged this from my site, please ref it back :)

```suggestion
/** https://github.com/juliusmarminge/jumr.dev/blob/main/app/og-image/get-fonts.ts */
export async function getFont<TWeights extends readonly number[]>({
```

(also you can use const params now in TS5)

Comment (iAverages) [reply www/src/utils/ogFont.ts:3]: I actually found it on the nextra branch that you send in Discord. I will add a link to your site though.

File: www/src/config.ts
Comment (c-ehrlich) [inline www/src/config.ts:391]: we already have this https://github.com/t3-oss/create-t3-app/blob/next/www/src/languages.ts#L13

**Review by c-ehrlich — COMMENTED** [review]
The OG images look really good now!

Would be nice to have a lil playground to be able to preview / modify them in dev. But IMO ok to merge this without, and open an issue to add it.

**Comment by iAverages** [comment]
> The OG images look really good now!
> 
> Would be nice to have a lil playground to be able to preview / modify them in dev. But IMO ok to merge this without, and open an issue to add it.

Glad you like them :)

Did you see my previous comment about the issues around enabling SSR with Astro? Is the sitemap something that is important to you guys? The sitemap plugin cannot automatically generate the sitemap for SSR routes and due to how the server output mode works, all routes are SSR routes by default.

**Review by c-ehrlich — APPROVED** [review]
Sorry for the delay, somehow missed this in my notifications

re SSR/sitemaps: let's solve that when we get to it :)

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**
